Data/models/DataSaved.swift

```diff
@@ -0,0 +1,52 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+
+public final class DataSaved: NSManagedObject, CRUD {
+    @NSManaged public var savedUrl: String
+    @NSManaged public var amount: String
+    
+    public class func get(with savedUrl: String) -> DataSaved? {
+        let predicate = NSPredicate(format: "\(#keyPath(DataSaved.savedUrl)) == %@", savedUrl)
+        return first(where: predicate)
+    }
+    
+    public class func all() -> [DataSaved] {
+        all() ?? []
```

Comment:
Recursive?

Reply:
No it is not, it is calling all method from Readable extension and If it is nil returning a empty list

```
    static func all(where predicate: NSPredicate? = nil,
                    sortDescriptors: [NSSortDescriptor]? = nil,
                    fetchLimit: Int = 0,
                    context: NSManagedObjectContext = DataController.viewContext) -> [Self]? {
        let request = getFetchRequest()
        
        request.predicate = predicate
        request.sortDescriptors = sortDescriptors
        request.fetchLimit = fetchLimit
        
        do {
            return try context.fetch(request)
        } catch {
            log.error("Fetch error: \(error)")
        }
        
        return nil
    }
```